routes/movies.routes.js

```diff
@@ -0,0 +1,94 @@
+const Movie = require("../models/Movie.model");
+const Celebrity = require("../models/Celebrity.model");
+const router = require("express").Router();
+
+
+router.get ('/', async (req,res) => {
+    try{
+        const allMovies = await Movie.find()
+    res.render("movies/movies", {allMovies})}   
+    catch(error){
+        console.log("error",error)
+    }
+    })
+
+
+router.get("/create",async (req,res) => {
+    try{
+        const movieCelebrities = await Celebrity.find();
+        res.render("movies/new-movie", {movieCelebrities})
+    }
+    catch(error){
+    console.log(error)
+    }
+        
+    })
+
+    router.get('/movies/create', (req, res) => {
+        res.render('movies/new-movie.hbs');
+    });    
+router.post("/create", async (req, res, next) => {
+    try{
+       const {title, plot, genre, cast} = req.body;
+    console.log(cast)  
+    await Movie.create({title, genre, plot, cast}).populate("cast")
+    res.redirect('/movies', { movie })
```

Comment:
You don't have to pass the { movie } to the redirect.
Maybe that's why it isn't working properly. It is creating the movie but not redirecting.